sdk/rust/oak/src/lib.rs

```diff
@@ -462,3 +461,69 @@ pub fn run_event_loop<T: crate::io::Decodable, N: Node<T>>(mut node: N, in_handl
         }
     }
 }
+
+/// Register a new node entrypoint.
```

Comment:
BTW, on first glance I found the `entrypoint!` name a smidgeon confusing, because the macro doesn't give a `name => entrypoint` mapping (for name = `ident`, entrypoint = `extern "C" fn(u64) -> ()`).  

It's more of a `name => constructor` mapping (for constructor = `fn () -> oak::Node<T> where T: io::Decodable`), but a macro named `entrypoint_constructor` isn't particularly euphonious, so maybe leave as-is unless you can think of the perfect name&hellip;

Reply:
I agree `entrypoint!` could be somewhat confusing to users that know what the actual entrypoint signature is. I can't think of a more descriptive name myself at the moment, I've also considered just calling it `node!`, but that probably only makes things more ambiguous.

Reply:
IMO it makes sense in that it creates an entrypoint (out of an expression returning a Node instance). Let's go ahead with this for now, we can rename if / when we come up with a better name.